website/docs/en/config/lib/auto-extension.mdx

```diff
@@ -3,4 +3,50 @@
 - **Type:** `boolean`
 - **Default:** `true`
 
-Whether to automatically set the file extension based on the `format` option in the JS and DTS output files.
+Whether to automatically set the file extension based on the [`format`](/config/lib/format) option in the JavaScript output files.
+
+## Default Extension
+
+By default that when `autoExtension` is set to `true`, the file extension will be:
+
+- `.js` with `esm` format and `.cjs` with `cjs` format when `type: module` in `package.json`.
+
+- `.js` with `cjs` format and `.mjs` with `esm` format when `type: commonjs` or no `type` field in `package.json`.
+
+:::warning
+
+When [`bundle`](/config/lib/bundle) is set to `false` that as known as bundleless mode, you should write full path instead of ignoring main file in source code.
```

Comment:
```suggestion
When [bundle](/config/lib/bundle) is set to `false` that as known as bundleless mode, you should write full path instead of ignoring directory indexes (e.g. `'./startup/index.js'`) in source code.
```

When a link is wrapped by code, the link underline will show in an inline code, which is weird. I checked Rsbuild's doc and found it doesn't wrap within code. We should follow that.

Reply:
And, should this section be moved to `bundle` or leave it in `auto-extension`? it's not relevant to extension IMO.

Reply:
Since we do not add `redirect.autoExtension`, I write here to do a stash.

Reply:
should `redirect.autoExtension` handle directory indexes?

Reply:
It is related to `resolve.mainFiles`, maybe need a `redirect.mainFiles` option